corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -302,3 +340,48 @@ def _apply_form_transaction(self, transaction):
             filtered_updates = [u for u in case_updates if u.id == self.case.case_id]
             for case_update in filtered_updates:
                 self._apply_case_update(case_update, form)
+
+
+def _transaction_sort_key_function(case):
+    def _transaction_cmp(first_transaction, second_transaction):
+        # if the forms aren't submitted by the same user, just default to server dates
+        if first_transaction.user_id != second_transaction.user_id:
+            return cmp(first_transaction.server_date, second_transaction.server_date)
+        else:
+            form_ids = list(case.xform_ids)
```

Comment:
not a big issue since this is memoized but could more this outside the function to avoid executing it for every comparison